src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -273,4 +269,41 @@ static Slice build(final Settings settings, final Authentication auth,
             .<Slice>map(limit -> new ContentLengthRestriction(slice, limit))
             .orElse(slice);
     }
+
+    /**
+     * Creates Docker proxy repository slice from configuration.
+     *
+     * @param cfg Repository configuration.
+     * @return Docker proxy slice.
+     */
+    private static Slice dockerProxy(final RepoConfig cfg) {
```

Comment:
@olegmoz I'd suggest to extract this method into separate class and properly unit test it